packages/core/src/foundation.tsx

```diff
@@ -70,12 +80,31 @@ class DdRumWrapper implements DdRumType {
 
     startAction(type: string, name: string, context: object = {}, timestampMs: number = timeProvider.now()): Promise<void> {
         InternalLog.log("Starting RUM Action “" + name + "” (" + type + ")", SdkVerbosity.DEBUG);
+        this.lastActionData={type, name}
         return this.nativeRum.startAction(type, name, context, timestampMs);
     }
 
-    stopAction(context: object = {}, timestampMs: number = timeProvider.now()): Promise<void> {
+    stopAction(...args: [type: string, name: string, context?: object, timestampMs?: number] | [context?: object, timestampMs?: number]): Promise<void> {
         InternalLog.log("Stopping current RUM Action", SdkVerbosity.DEBUG);
-        return this.nativeRum.stopAction(context, timestampMs);
+        let returnPromise = this.getStopActionNativeCall(args)
+        this.lastActionData = undefined
+        return returnPromise
+    }
+
+    private getStopActionNativeCall = (args: [type: string, name: string, context?: object, timestampMs?: number] | [context?: object, timestampMs?: number]): Promise<void> => {
+        if (isNewStopActionAPI(args)) {
+            return this.nativeRum.stopAction(args[0], args[1], args[2] || {}, args[3] || timeProvider.now())
+        }
+        if (isOldStopActionAPI(args)) {
+            if (this.lastActionData) {
+                return this.nativeRum.stopAction(this.lastActionData.type, this.lastActionData.name, args[0] || {}, args[1] || timeProvider.now());
+            } 
+            InternalLog.log("DdRum.startAction needs to be called before DdRum.stopAction", SdkVerbosity.WARN);
```

Comment:
Here I think it's too complex to replicate the way the SDKs work for edge cases: when we start 2 actions in parallel for instance. The Android SDK is handling it differently from the iOS bridge:
- https://github.com/DataDog/dd-sdk-android/blob/72beeb9ffdc3aa08e4e7128bf3f67a11a14a90b2/dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt#L265
- https://github.com/DataDog/dd-bridge-ios/blob/main/DatadogSDKBridge/Classes/Implementation/DdRumImplementation.swift#L123

Let's go with this very simple implementation and then remove this API

Reply:
so basically you rely on the native SDK to handle the `stopAction` properly ? I guess it could work, but you will loose the console logs. We had situations where users did not understand why their actions were not visible.

Reply:
What do you mean by I would lose the console logs?
I have still a console.log in `stopAction` to inform that the action has been executed, and there will be a warn in case the event won't be treated by the native SDK

Reply:
I mean that there won't be a case where we call make the call to the native SDK and then the native SDK rejects the action - at least with the old API.
Doing it for the new API would require more time to handle edge cases such as parallel actions.